test-system/src/lib.rs

```diff
@@ -135,8 +138,55 @@ macro_rules! test_runtest {
         test_declare!($n);
         let res: TestResult = $b;
         match res {
-            TestResult::Pass => test_pass!($n),
-            TestResult::Fail => test_fail!($n),
+            Ok(()) => test_pass!($n),
+            Err(TestFailure::Fail) => test_fail!($n),
+            Err(TestFailure::FailedAt(s)) => test_fail!($n, s),
+        }
+    };
+}
+
+/// Test if an expression is true, with a name for the test
+/// and a boolean to carry results from multiple test
+/// ```
+/// use test_system::*;
+/// let results = true;
+/// test_result_true(1 == 1, "first", results);
+/// test_result_true(2 == 2, "second", results);
+///
+/// ```
+#[macro_export]
+macro_rules! test_result_true {
+    ($e:expr, $n: expr, $f:expr) => {
+        test_declare!($n);
+        let res: bool = $e;
+        $f = $f && res;
+        if res {
+            test_pass!($n);
+        } else {
+            test_fail!($n)
+        }
+    };
+}
+
+/// Test if an expression is false, with a name for the test
+/// and a boolean to carry results from multiple test
+/// ```
+/// use test_system::*;
+/// let results = true;
```

Comment:
Nit: Doc test will fail because results needs to mutable: let mut results = true;

Reply:
fixed